boot/freeldr/freeldr/arch/uefi/uefimem.c

```diff
@@ -0,0 +1,309 @@
+/*
+ * PROJECT:     FreeLoader UEFI Support
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Memory Managment Functions
+ * COPYRIGHT:   Copyright 2022 Justin Miller <justinmiller100@gmail.com>
+ */
+
+/* INCLUDES ******************************************************************/
+
+#include <uefildr.h>
+
+#include <debug.h>
+DBG_DEFAULT_CHANNEL(WARNING);
+
+#define NEXT_MEMORY_DESCRIPTOR(Descriptor, DescriptorSize) \
+    (EFI_MEMORY_DESCRIPTOR*)((char*)(Descriptor) + (DescriptorSize))
+#define EXIT_STACK_SIZE 0x1000
+#define UNUSED_MAX_DESCRIPTOR_COUNT 10000
+
+ULONG
+AddMemoryDescriptor(
+    _Inout_ PFREELDR_MEMORY_DESCRIPTOR List,
+    _In_ ULONG MaxCount,
+    _In_ PFN_NUMBER BasePage,
+    _In_ PFN_NUMBER PageCount,
+    _In_ TYPE_OF_MEMORY MemoryType);
+
+/* GLOBALS *******************************************************************/
+
+extern EFI_SYSTEM_TABLE * GlobalSystemTable;
+extern EFI_HANDLE GlobalImageHandle;
+extern REACTOS_INTERNAL_BGCONTEXT framebufferData;
+
+EFI_MEMORY_DESCRIPTOR* EfiMemoryMap = NULL;
+UINT32 FreeldrDescCount;
+PVOID OsLoaderBase;
+SIZE_T OsLoaderSize;
+EFI_HANDLE PublicBootHandle;
+PVOID ExitStack;
+PVOID EndofExitStack;
```

Comment:
Is it going to be used elsewhere in the code later?

Reply:
Yes this is required for the exit code.